src/main/java/com/google/api/codegen/nodejs/JSDocCommentFixer.java

```diff
@@ -0,0 +1,94 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.nodejs;
+
+import com.google.api.codegen.CommentPatterns;
+import com.google.common.base.Splitter;
+
+import java.util.regex.Matcher;
+
+/**
+ * Utility class for formatting source comments to follow JSDoc style.
+ */
+public class JSDocCommentFixer {
+
+  /**
+   * Returns a JSDoc-formatted comment string.
+   */
+  public static String jsdocify(String comment) {
+    comment = jsdocifyProtoMarkdownLinks(comment);
+    comment = jsdocifyCloudMarkdownLinks(comment);
+    return cleanupTrailingWhitespaces(comment);
+  }
+
+  private static String protoToRubyDoc(String comment) {
+    boolean messageFound = false;
+    boolean isFirstSegment = true;
+    String result = "";
+    for (String name : Splitter.on(".").splitToList(comment)) {
+      char firstChar = name.charAt(0);
+      if (Character.isUpperCase(firstChar)) {
+        messageFound = true;
+        result += (isFirstSegment ? "" : "::") + name;
+      } else if (messageFound) {
+        // Lowercase segment after message is found is field.
+        // In Ruby, it is referred as "Message#field" format.
+        result += "#" + name;
+      } else {
+        result +=
+            (isFirstSegment ? "" : "::") + Character.toUpperCase(firstChar) + name.substring(1);
+      }
+      isFirstSegment = false;
+    }
+    return result;
+  }
+
+  /**
+   * Returns a string with all proto markdown links formatted to JSDoc style.
+   */
+  private static String jsdocifyProtoMarkdownLinks(String comment) {
+    StringBuffer sb = new StringBuffer();
+    Matcher m = CommentPatterns.PROTO_LINK_PATTERN.matcher(comment);
+    if (!m.find()) {
+      return comment;
+    }
+    do {
+      m.appendReplacement(sb, String.format("{@link %s}", m.group(1)));
+    } while (m.find());
+    m.appendTail(sb);
+    return sb.toString();
+  }
+
+  /**
+   * Returns a string with all cloud markdown links formatted to JSDoc style.
+   */
+  private static String jsdocifyCloudMarkdownLinks(String comment) {
+    StringBuffer sb = new StringBuffer();
+    Matcher m = CommentPatterns.CLOUD_LINK_PATTERN.matcher(comment);
+    if (!m.find()) {
+      return comment;
+    }
+    do {
+      String url = "https://cloud.google.com" + m.group(2);
+      m.appendReplacement(sb, String.format("[%s](%s)", m.group(1), url));
+    } while (m.find());
+    m.appendTail(sb);
+    return sb.toString();
+  }
+
+  private static String cleanupTrailingWhitespaces(String comment) {
```

Comment:
Why does this need to be a separate method?

(I see that this probably comes from Ruby, but there too, this method seems only to wrap `trim()`)